values.yaml

```diff
@@ -1004,6 +1004,59 @@ meshGateway:
   #     "annotation-key": "annotation-value"
   annotations: null
 
+ingressGateways:
+  enabled: true
+
+  # Defaults sets default values for all gateway fields. With the exception
+  # of annotations, defining any of these values in the `gateways` list
+  # will override the default values provided here.
+  defaults:
+    replicas: 2
+    imageEnvoy: "envoyproxy/envoy:v1.13.0"
+
+    wanAddress:
+      source: "Service"
+      port: 443
+      static: ""
+
+    hostPort: null
```

Comment:
If folks want to expose their ingress gateways on a node to avoid the 2 extra hops (load balancer -> random node in cluster -> ingress pod vs  -> ingress pod) then they'd need this.

We can remove it and add it later but if it's already coded up in mesh gateways and so doesn't cost us any more to implement it *and* it makes the gateway configs consistent then I think we should keep it.

Reply:
I think we should just recommend service type NodePort for those cases. I believe it's the same number of hops for service type NodePort vs hostport.

I'm just thinking that if we don't have a compelling use case for host port that wouldn't be also solved by service type NodePort, then I think it makes sense to avoid including a configuration that's explicitly not recommended in kube docs.

Reply:
> I believe it's the same number of hops for service type NodePort vs hostport.

Can you double check that coming in through the nodeport guarantees you route to the pod on that host?

Reply:
I'm pretty sure it just hits the kubeproxy iptables and gets randomly forwarded to one of the pod ips

Reply:
Oh, I see what you mean. Yeah, you're right, by default kubeproxy will pick a backend pod at random, which could add an extra hop. Although you can [configure](https://kubernetes.io/docs/concepts/services-networking/service/#proxy-mode-ipvs) the kubeproxy to run in a different mode. This configurability likely is not supported by most cloud providers.

I think then if we wanted to support the use case you're describing we'd need to support host port.